### PR TITLE
expose full user id config (including storage) to user id modules

### PR DESCRIPTION
**Pull request by smenzer**
## Type of change
- [ x ] Feature

## Description of change
Today, user id modules only receive the `params` object from publisher configuration. However, the `storage` object can also be useful to allow the submodule to know the storage type / name being used, what the refresh settings are, etc. To accomplish this, we have to pass the entire config object to the submodule in `getId`, `extendId`, and `decode`. 

This PR also updates every submodule and their tests to ensure they are still working as they were previously.

### CODE REVIEW

**Review by jlukas79 — APPROVED** [review]

**Comment by smenzer** [comment]
@harpere what docs do you envision adding here? the docs website doesn't mention anything about how to build a user id module and i've updated the relevant comment docs within the user id module code. happy to provide more, just not sure where this would be documented.

**Review by FilipStamenkovic — APPROVED** [review]
LGTM.  
I tested it on my local, works fine. Just one small question:

You changed from `@param {SubmoduleParams} [submoduleConfigParams]` to `@param {SubmoduleParams} [submoduleConfig]` in every submodule. More specifically, you changed `submoduleConfigParams` to `submoduleConfig`. Why didn't you change `SubmoduleParams` to `SubmoduleConfig` also?

**Comment by smenzer** [comment]
> You changed from `@param {SubmoduleParams} [submoduleConfigParams]` to `@param {SubmoduleParams} [submoduleConfig]` in every submodule. More specifically, you changed `submoduleConfigParams` to `submoduleConfig`. Why didn't you change `SubmoduleParams` to `SubmoduleConfig` also?

No reason really...I can do that now.

**Comment by smenzer** [comment]
@FilipStamenkovic done...if you're ok with this, can you merge, please? thanks!

File: modules/britepoolIdSystem.js
Comment (FilipStamenkovic) [inline modules/britepoolIdSystem.js:88]: I don't think you should have changed this one?

File: modules/liveIntentIdSystem.js
Comment (FilipStamenkovic) [inline modules/liveIntentIdSystem.js:103]: Here `SubmoduleConfig` instead of `SubmoduleParams`?

File: modules/userId/index.js
Comment (FilipStamenkovic) [inline modules/userId/index.js:30]: Here `SubmoduleConfig` instead of `SubmoduleParams`?

File: modules/userId/index.js
Comment (FilipStamenkovic) [inline modules/userId/index.js:40]: also here?

**Review by FilipStamenkovic — CHANGES_REQUESTED** [review]
I found couple of places where `SubmoduleParams` was not changed to `SubmoduleConfig`.  

What is the status with docs? I see @harpere requested some docs, is this still valid? I'm asking because if some docs are needed then I cannot merge this PR?

File: modules/britepoolIdSystem.js
Comment (smenzer) [reply modules/britepoolIdSystem.js:88]: you're right

File: modules/liveIntentIdSystem.js
Comment (smenzer) [reply modules/liveIntentIdSystem.js:103]: yep

File: modules/userId/index.js
Comment (smenzer) [reply modules/userId/index.js:30]: yep, i forgot i changed extendId too, good catch

**Review by smenzer — COMMENTED** [review]
good finds...updating again now

File: modules/userId/index.js
Comment (smenzer) [reply modules/userId/index.js]: yea

**Comment by smenzer** [comment]
happy to provide docs, but like i mentioned before, I'm not sure where they belong. we don't really have docs on how to write a user id module AFAIK

**Review by FilipStamenkovic — APPROVED** [review]

**Comment by FilipStamenkovic** [comment]
> happy to provide docs, but like i mentioned before, I'm not sure where they belong. we don't really have docs on how to write a user id module AFAIK

@harpere Do we need some docs, or can we merge this one?

**Comment by smenzer** [comment]
I'd suggest we merge this and if @harpere identifies where we should add docs, I can follow up with a PR right away. i'd like to get this in before the other outstanding PRs for user id modules get merged to avoid having to rework those here.

since there are two approvals and tests are passing, i'll give it a few hours in case anyone objects, but then i'll go ahead and merge this.

cc @FilipStamenkovic

**Pull request merged**
